### DISCUSSION TITLE
[Bug] Apply Tag Rules does not working properly with sad panda tags.

### DISCUSSION THREAD

**Discussion by PasserDreamer** [Difegue/LANraragi#592] [Q&A] [answered]
**LRR Version and OS**
Docker Version 0.8.4

**Bug Details**
Single tag doesn't work, only tag with namespace work.
Other tags like "already uploaded" don't work either because of missing namespace "other:"

**Screenshots**
![123](https://user-images.githubusercontent.com/30385417/148532571-b6f13512-2df2-47c1-a6d5-eb776c4dc127.png)

**Comment by IceBreeze** [comment, marked as answer]
I don't see the problem, it seems to work as expected. The tag returned by the API is:

```txt
other:already uploaded
```
So the rule should be:

```txt
-other:already uploaded
```

`other:` is a namespace on par with `female:` and I don't feel like treating it differently.
However, due to the fact that the rules are applied sequentially to every tag, there is a small undocumented trick you can use if you want to treat `other:` like a "virtual" namespace:

```txt
~other
-already uploaded
```

This sequence of rules will strip the namespace (from all the tags with that namespace) and then it will remove the tag so you won't have to write:

```txt
-other:already uploaded
-already uploaded
```

The same goes for the substitution. You can't write the rule:

```txt
mind control -> MC
```

and expect it to work to replace the string "mind control" regardless of the namespace like in this example:

```txt
mind control          ---> MC
female:mind control   ---> female:MC
foo:mind control      ---> foo:MC
bar:mind control      ---> bar:MC
xpto:mind control     ---> xpto:MC
```

If you want a rule to do that, it should be something like `*:tag -> new tag`, but I doesn't seem so useful compared to the combination of the other rules, or maybe I didn't understand what you are expecting.

**Comment by Difegue** [comment]
Looks like expected behavior indeed on that one -- Converting to a Q&A discussion. Thanks for clarifying, @IceBreeze!
